avocado/core/future/settings.py

```diff
@@ -0,0 +1,364 @@
+    """
+
+    def __init__(self, path_list):
+        super(ConfigFileNotFound, self).__init__()
+        self.path_list = path_list
+
+    def __str__(self):
+        return ("Could not find the avocado config file after looking in: %s" %
+                self.path_list)
+
+
+class AvocadoSettings:
+    """AvocadoSettings, an experimental Avocado configuration handler.
+
+    It is a simple wrapper around configparser and argparse.
+
+    Also, one object of this class could be passed as config to plugins and
+    modules.
+
+    Please, not that most of methods and attributes here are private. Only
+    public methods and attributes should be used outside this module.
+    """
+
+    def __init__(self, config_path=None):
+        """Constructor. Tries to find the main settings files and load them.
+
+        :param config_path: Path to a config file. Useful for unittesting.
+        """
+        self._config = configparser.ConfigParser()
+        self._all_config_paths = []
+        self._config_paths = []
+        self._short_mapping = {}
+        self._long_mapping = {}
+        self._namespaces = {}
+
+        # 1. Prepare config paths
+        if config_path is None:
+            self._prepare_base_dirs()
+            self._append_config_paths()
+        else:
+            # Only used by unittests (the --config parses the file later)
+            self._all_config_paths.append(config_path)
+
+        # 2. Parse/read all config paths
+        self._config_paths = self._config.read(self._all_config_paths)
+        if not self._config_paths:
+            raise ConfigFileNotFound(self._all_config_paths)
+
+    def _append_config_paths(self):
+        # 1. Append Defaults
+        self._append_pkg_defaults()
+
+        # 2. Override with system config
+        self._append_system_config()
+
+        # Allow plugins to modify/extend the list of configs
+        dispatcher = SettingsDispatcher()
+        if dispatcher.extensions:
+            dispatcher.map_method('adjust_settings_paths',
+                                  self._all_config_paths)
+
+        # 3. Override with the user's local config
+        self._append_user_config()
+
+    def _append_pkg_defaults(self):
+        config_pkg_base = os.path.join('etc', 'avocado', 'defaults.conf')
+        config_path_pkg = resource_filename('avocado', config_pkg_base)
+        self._all_config_paths.append(config_path_pkg)
+
+    def _append_system_config(self):
+        self._all_config_paths.append(self._config_path_system)
+        configs = glob.glob(os.path.join(self._config_dir_system_extra,
+                                         '*.conf'))
+        for extra_file in configs:
+            self._all_config_paths.append(extra_file)
+
+    def _append_user_config(self):
+        if not os.path.exists(self._config_path_local):
+            self._create_empty_config()
+        self._all_config_paths.append(self._config_path_local)
+
+    def _create_empty_config(self):
```

Comment:
This is not related to your code, but I have the feeling that we should avoid trying to create configuration files at runtime.  As this code shows, it's error prone, and IMO its value can be substituted by good documentation.  If you agree, we can remove that behavior ASAP from the current `avocado.core.settings` implementation and add documentation instead.

Reply:
I agree with that, but I think that we should focus on `future.settings` instead. So if you are ok, I will remove this on a different PR.

Reply:
Yep, as I said, it's not related at all.  Created an issue to track it:

https://github.com/avocado-framework/avocado/issues/3541